test/e2e/reporters/progress.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reporters
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/types"
+	"k8s.io/klog"
+)
+
+// ProgressReporter is a ginkgo reporter which tracks the total number of tests to be run/passed/failed/skipped.
+// As new tests are completed it updates the values and prints them to stdout and optionally, sends the updates
+// to the configured URL.
+type ProgressReporter struct {
+	LastMsg string `json:"msg"`
+
+	NumToBeRun   int `json:"total"`
```

Comment:
Thoughts on `TestsTotal`?

Reply:
The naming of total/failed/skipped all comes from the desired integration with Sonobuoy which has already defined these fields.

Reply:
Okay, figured it's fine to change it here since we only care about the json tags

Reply:
Oh I wa shrinking you meant the json tags; I’ll change the names of the variable Lea that’s fine.